File: cpp/Flashcard.cpp
#include "Flashcard.h"

// Default Constructor
Flashcard::Flashcard(){}

// Overload Constructor
Flashcard::Flashcard(string front, string back) {
    this->front = front;
    this->back = back;
    difficulty = 1;
}

// Overload Constructor
Flashcard::Flashcard(string front, string back, int difficulty) {
    this->front = front;
    this->back = back;
    this->difficulty = difficulty;
}

// Deconstructor
Flashcard::~Flashcard(){}

// Setters
void Flashcard::setFront(string Front) {
    this->front = Front;
}
void Flashcard::setBack(string Back) {
    this->back = Back;
}
void Flashcard::setDifficulty(int difficulty) {
    this->difficulty = difficulty;
}

// Getters
string Flashcard::getFront() const {
    return front;
}
string Flashcard::getBack() const {
    return back;
}
int Flashcard::getDifficulty() const {
    return difficulty;
}

File: java/Flashcard.java
public class Flashcard {
    String front;
    String back;

    public Flashcard() {
        this.front = "";
        this.back = "";
    }
    
    public Flashcard(String front, String back) {
        this.front = front;
        this.back = back;
    }

    // Getters 
    public String getFront() {
        return front;
    }

    public String getBack() {
        return back;
    }

    // Setters
    public void setFront(String front) {
        this.front = front;
    }

    public void setBack(String back) {
        this.back = back;
    }

    // toString method for better printing
    @Override
    public String toString() {
        return "Front: " + front + " | Back: " + back;
    }

}
